client/src/components/chess-game/ChessGame.js

```diff
@@ -98,16 +109,27 @@ export default class ChessGame extends React.Component {
   };
 
   startGame() {
-    this.clearTimers();
+    createConnection();
     this.chess = new Chess();
     this.squares = this.chess.SQUARES;
     const board = this.chess.board();
+
     const currentPlayer = this.chess.turn();
-    this.setState(Object.assign({}, INIT_STATE, { board, currentPlayer }));
+    receiveGameData(({ game }) => {
+      let { playerColor } = this.state;
+      if (!playerColor && !game.player2) {
+        playerColor = game.player1.color;
+      }
+      if (!playerColor && game.player2) {
+        playerColor = game.player2.color;
+      }
+      this.setState(Object.assign({}, INIT_STATE, { board, currentPlayer, playerColor }));
+    });
   }
 
   movePiece(from, to) {
     this.chess.move({ from, to });
+    sendMove({ from, to });
```

Comment:
is this non blocking call?
We may not want to block UI while socket is sending data. And have the bord also update immediately.

Reply:
This is a non-blocking call.